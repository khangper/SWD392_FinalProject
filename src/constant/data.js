export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
