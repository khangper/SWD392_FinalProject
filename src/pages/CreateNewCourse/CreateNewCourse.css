/* CreateNewCourse.css */

.create-new-course-container {
  display: flex;
  flex-direction: column;
  background-color: #f7f7f7;
  padding: 20px;
}

.create-new-course-header {
  width: 100%;
}

.create-new-course-title {
  font-size: 20px;
  margin-top: 0 !important;
  margin-bottom: 8px !important;
  font-weight: bold;
  font-family: 'Roboto', sans-serif;
  color: #333;
}

.create-new-course-title img {
  width: 20px;
  height: 20px;
  margin-right: 10px;
  position: relative;
  top: 3px;
}

.create-new-course-body {
  margin-top: 60px;
}

.step-steps {
  display: flex;
  justify-content: space-between;
  position: relative;
  padding: 0;
  list-style: none;
}

.step-steps li {
  position: relative;
  text-align: center;
  flex: 1;
}

.step-steps li:not(:first-child)::before {
  content: '';
  position: absolute;
  left: -50%;
  top: 30%;
  width: 100%;
  height: 8px;
  background-color: #ededed;
  transition: all 0.3s;
}

.step-steps li.done:not(:first-child)::before,
.step-steps li.active:not(:first-child)::before {
  background-color: #FFE599;
}

.step-steps li a {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-decoration: none;
}

.step-steps li .number {
  background-color: #dae1e8;
  border: 1px solid #b6c2cf;
  border-radius: 50%;
  color: #454e5b;
  font-size: 18px;
  font-weight: bold;
  line-height: 50px;
  margin: 10px auto;
  position: relative;
  width: 24px;
  height: 24px;
  z-index: 1;
}

.step-steps li.active .number,
.step-steps li.done .number {
  background-color: #F9BF4B;
  box-shadow: 0 0 0 5px rgba(237, 42, 38, 0.3);
}

.step-steps li .step-name {
  color: #454e5b;
  font-size: 14px;
  margin-top: 5px;
}

.step-steps li.active .step-name,
.step-steps li.done .step-name {
  color: #196c7e;
}

.content-section {
  padding-top: 60px;
}

.content-tab-header {
  display: flex;
  width: 100%;
  border-bottom: 1px solid #efefef;
  border-top: 1px solid #efefef;
  padding: 18px 0;
  margin-bottom: 25px;
}

.content-tab-header h3 {
  display: flex;
  font-size: 18px;
  font-family: 'Roboto', sans-serif;
  font-weight: bold;
  margin-bottom: 0;
  text-transform: capitalize;
  align-items: center;
}

.content-tab-header h3 img {
  width: 25px;
  height: 25px;
  margin-right: 10px;
}

.content-tab-content {
  background: #fff;
  margin-top: 30px;
  padding: 0 30px 30px 30px;
  border-radius: 3px;
  border: 1px solid #efefef;
}

.content-tab-title {
  margin-top: 32px;
}

.content-tab-title h4 {
  font-size: 14px;
  font-weight: bold;
  font-family: 'Roboto', sans-serif;
  margin-bottom: 10px !important;
  color: #333;
  text-align: left;
  display: block;
}

.content-tab-form {
  display: flex;
  flex-direction: column;
}

.content-tab-form .headline-group {
  width: 100%;
  margin-top: 0;
}

.content-tab-form .headline-group .form-group .form-counter {
  background: #196c7e;
  color: #fff;
  font-weight: bold;
  font-family: 'Roboto', sans-serif;
  position: absolute;
  right: 3px;
  border-radius: 3px;
  height: 34px;
  top: 3px;
  width: 34px;
  font-size: 14px;
  padding: 8px 9px;
}

.content-tab-form .headline-group .form-group textarea {
  resize: vertical;
  height: 130px;
  width: 100%;
  padding: 10px;
  background: #fff;
  border: 1px solid #e5e5e5 !important;
  outline: 0;
  color: rgba(0, 0, 0, .87);
  border-radius: 3px !important;
  box-shadow: inset 0px 1px 4px rgba(0, 0, 0, 0.02) !important;
}

.ck-editor__editable_inline {
  min-height: 150px;
}

.create-new-course-footer {
  margin-top: 20px;
}

.content-tab-form-split {
  display: flex;
  width: 100%;
  gap: 20px;
}

.content-tab-form-left {
  flex: 1;
  margin-top: 0;
}

.content-tab-form .form-group .help-block {
  position: relative;
  font-size: 12px;
  line-height: 20px;
  font-family: Roboto;
  color: #686f7a;
  width: 100%;
}

.content-tab-form .form-group {
  position: relative;
}

.content-tab-form .form-group .react-select-container {
  width: 100%;
}

.content-tab-form .form-group .react-select__control {
  padding: 5px;
  font-size: 14px;
  border: 1px solid #ccc;
  border-radius: 4px;
  background-color: #fff;
  height: 40px;
}

.content-tab-form .form-group .react-select__menu {
  font-size: 1rem;
}

.content-tab-form .form-group .react-select__multi-value__label {
  color: #00b2a9;
}
.step-footer {
  display: flex;
  justify-content: space-between;
  margin-top: 30px;
}

.steps_btn {
  background-color: #fff;
  border: 1px solid #f9f9f9;
  color: #00b2a9;
  padding: 10px 20px;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.3s, color 0.3s;
}

.steps_btn:hover {
  background-color: gray;
  color: #fff;
}

.steps_btn:disabled {
  background-color: #f7f7f7;
  color: #ccc;
  cursor: not-allowed;
}
.curriculum-tab-content {
  margin-top: 30px;
}

.curriculum-add-item {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  margin-top: 5px;
  border: 1px solid #efefef;
  background: #fff;
  padding: 15px 20px;
}

.curriculum-add-item h4 {
  margin-bottom: 0;
  font-size: 18px;
  color: #333;
  display: flex;
  align-items: center;
}

.curriculum-add-item h4 img {
  width: 25px;
  height: 25px;
  margin-right: 5px;
}

.curriculum-btn {
  height: 40px;
  padding: 0 20px;
  font-size: 14px;
  color: #fff;
  border: 0;
  background: #196c7e;
  border-radius: 3px;
  margin-right: 5px;
  font-weight: 500;
  font-family: "Roboto", sans-serif !important;
  cursor: pointer;
}
.curriculum-btn:hover{
  background-color: #0d3842;
  color: #fff;
}
.added-section-item{
  border: 1px solid #efefef;
    margin-top: 30px;
    background: #ffffff;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}
.added-section-header {
  background: #ffffff;
  padding: 15px 20px;
  border-radius: 3px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: -1px;
  border-bottom: 1px solid #efefef;
  width: 100%;
}

.section-title {
  display: flex;
  align-items: center;
}

.section-title h4 {
  margin-bottom: 0;
  font-size: 16px;
  font-weight: bold;
  color: #333;
  display: flex;
  align-items: center;
}

.section-title h4 img {
  width: 15px;
  height: 15px;
  margin-right: 5px;
}

.section-edit-options {
  display: flex;
  align-items: center;
}

.section-edit-options img {
  width: 20px;
  height: 20px;
  margin-left: 10px;
  cursor: pointer;
}
.section-item h4 img {
  width: 15px;
  height: 15px;
  margin-right: 5px;
}
.section-group-sortable-list {
  margin-top: 20px;
}

.section-item {
  background: #f8f8f8;
  padding: 15px 20px;
  border: 1px solid #efefef;
  border-radius: 3px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  cursor: move;
}

.section-item h4 {
  margin-bottom: 0;
  font-size: 14px;
  color: #333;
  display: flex;
  align-items: center;
}

.section-item h4 img {
  width: 15px;
  height: 15px;
  margin-right: 5px;
}

.section-item:hover {
  background: #ececec;
}
.section-add-item-wrap{
    background: #231f20;
    color: #fff;
    padding: 1rem !important;
    display: flex;
    gap: 10px;
}
.section-add-btn {
  display: flex;
  align-items: center;
  background-color: transparent;
  color: #fff;
  border: none;
  font-size: 16px;
  cursor: pointer;
}

.section-add-btn img {
  margin-right: 5px;
  width: 16px;
  height: 16px;
}
.video-content-tab{
  display: flex;
    flex-direction: column;
    margin-top: 30px;
}
.video-content-tab .video-info{
  font-size: 14px;
  font-family: 'Roboto', sans-serif;
  margin-bottom: 11px;
  display: block;
  color: #686f7a;
}
.html5-left-course{
  flex: 0 0 auto;
  width: 41.66666667%;
}
.create-new-course-video-body{
  flex: 0 0 auto;
  width: 41.66666667%;
}
.thumb-item{
  width: 100%;
  border: 1px solid #efefef;
  text-align: center;
  border-radius: 3px;
  margin-bottom: 0;
  background: #fff;
}
.thumb-item img{
  width: 100%;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
}
.thumb-dt{
  padding: 20px 20px 17px;
}
.publish-block{
  padding: 30px;
    text-align: center;
    border: 1px solid #efefef;
    background: #ffffff;
    border-radius: 3px;
    margin-top: 30px;
}
.publish-block img{
  height: 30px;
  width: 30px;
}
.publish-block p{
  font-size: 14px;
  line-height: 26px;
  padding-top: 20px;
  margin-bottom: 0;
  color: #686f7a;
}