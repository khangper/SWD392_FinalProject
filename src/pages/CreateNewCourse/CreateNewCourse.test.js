import React from "react";
import { render, screen } from "@testing-library/react";
import CreateNewCourse from "../CreateNewCourse/CreateNewCourse";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";
import userEvent from "@testing-library/user-event";
describe("CreateNewCourse", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the CreateNewCourse component", async () => {
    const { container } = renderWithProviders(<CreateNewCourse />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  // Course title here

  it("handles whitespace and special characters in the title input", async () => {
    renderWithProviders(<CreateNewCourse />);

    const titleInput = screen.getByPlaceholderText("Course title here");

    // Simulate entering whitespace
    userEvent.type(titleInput, "   ");
    expect(titleInput.value).toBe("   ");

    // Simulate entering special characters
    userEvent.clear(titleInput);
    userEvent.type(titleInput, "!@#$%^&*()_+");
    expect(titleInput.value).toBe("!@#$%^&*()_+");

    // Simulate entering a mix of whitespace and special characters
    userEvent.clear(titleInput);
    userEvent.type(titleInput, "   !@#$%^&*()_+   ");
    expect(titleInput.value).toBe("   !@#$%^&*()_+   ");
  });

  // Author here
  it("handles text input in the author textarea", async () => {
    renderWithProviders(<CreateNewCourse />);

    const authorTextarea = screen.getByPlaceholderText("Author here...");

    // Simulate entering text
    userEvent.type(authorTextarea, "John Doe");
    expect(authorTextarea.value).toBe("John Doe");

    // Simulate entering whitespace
    userEvent.clear(authorTextarea);
    userEvent.type(authorTextarea, "   ");
    expect(authorTextarea.value).toBe("   ");

    // Simulate entering special characters
    userEvent.clear(authorTextarea);
    userEvent.type(authorTextarea, "!@#$%^&*()_+");
    expect(authorTextarea.value).toBe("!@#$%^&*()_+");

    // Simulate entering a mix of whitespace and special characters
    userEvent.clear(authorTextarea);
    userEvent.type(authorTextarea, "   !@#$%^&*()_+   ");
    expect(authorTextarea.value).toBe("   !@#$%^&*()_+   ");
  });
});
