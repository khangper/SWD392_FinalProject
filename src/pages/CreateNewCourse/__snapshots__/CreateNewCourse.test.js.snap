// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewCourse renders the CreateNewCourse component 1`] = `
<div>
  <div
    class="create-new-course-container"
  >
    <div
      class="create-new-course-header"
    >
      <h2
        class="create-new-course-title"
      >
        <img
          alt="Create Course Icon"
          src="browse-instructor.png"
        />
        Create a New Course
      </h2>
    </div>
    <div
      class="create-new-course-body"
    >
      <ul
        class="step-steps"
      >
        <li
          class="step-item active"
        >
          <a
            class="flex"
            href="#tab_step1"
          >
            <span
              class="number"
            />
            <span
              class="step-name"
            >
              Basic
            </span>
          </a>
        </li>
        <li
          class="step-item "
        >
          <a
            class="flex"
            href="#tab_step2"
          >
            <span
              class="number"
            />
            <span
              class="step-name"
            >
              Curriculum
            </span>
          </a>
        </li>
        <li
          class="step-item "
        >
          <a
            class="flex"
            href="#tab_step3"
          >
            <span
              class="number"
            />
            <span
              class="step-name"
            >
              Media
            </span>
          </a>
        </li>
        <li
          class="step-item "
        >
          <a
            class="flex"
            href="#tab_step4"
          >
            <span
              class="number"
            />
            <span
              class="step-name"
            >
              Price
            </span>
          </a>
        </li>
        <li
          class="step-item "
        >
          <a
            class="flex"
            href="#tab_step5"
          >
            <span
              class="number"
            />
            <span
              class="step-name"
            >
              Publish
            </span>
          </a>
        </li>
      </ul>
      <section
        class="content-section"
      >
        <div
          class="content-tab-container"
        >
          <div
            class="content-tab-header"
          >
            <h3>
              <img
                src="info.png"
              />
              Basic Information
            </h3>
          </div>
          <div
            class="content-tab-content"
          >
            <div
              class="content-tab-title"
            >
              <h4>
                Course title*
              </h4>
            </div>
            <div
              class="content-tab-form"
            >
              <div
                class="headline-group"
              >
                <div
                  class="form-group"
                >
                  <input
                    id="title"
                    name="title"
                    placeholder="Course title here"
                    type="text"
                    value=""
                  />
                  <div
                    class="form-counter"
                  >
                    60
                  </div>
                </div>
                <div
                  class="help-block"
                >
                  (Please make this a maximum of 100 characters and unique.)
                </div>
              </div>
              <div
                class="content-tab-title"
              >
                <h4>
                  Author*
                </h4>
              </div>
              <div
                class="headline-group"
              >
                <div
                  class="form-group"
                >
                  <textarea
                    id="author"
                    name="author"
                    placeholder="Author here..."
                    type="text"
                  />
                  <div
                    class="help-block"
                  >
                    220 words.
                  </div>
                </div>
              </div>
              <div
                class="content-tab-title"
              >
                <h4>
                  Course description*
                </h4>
              </div>
              <div
                class="headline-group"
              >
                <div />
              </div>
              <div
                class="content-tab-form-split"
              >
                <div
                  class="content-tab-form-left"
                >
                  <div
                    class="content-tab-title"
                  >
                    <h4>
                      Date*
                    </h4>
                  </div>
                  <div
                    class="headline-group"
                  >
                    <div
                      class="form-group"
                    >
                      <textarea
                        id="date"
                        name="date"
                        type="text"
                      />
                      <div
                        class="help-block"
                      >
                        Student will gain these skills and knowledge after completing this course. (One per line).
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="content-tab-form-left"
                >
                  <div
                    class="content-tab-title"
                  >
                    <h4>
                      Hours*
                    </h4>
                  </div>
                  <div
                    class="headline-group"
                  >
                    <div
                      class="form-group"
                    >
                      <textarea
                        id="hours"
                        name="hours"
                        type="text"
                      />
                      <div
                        class="help-block"
                      >
                        What knowledge, technology, and tools are required by users to start this course. (One per line).
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="content-tab-form-split"
              >
                <div
                  class="content-tab-form-left"
                >
                  <div
                    class="content-tab-title"
                  >
                    <h4>
                      Course level*
                    </h4>
                  </div>
                  <div
                    class="headline-group"
                  >
                    <div
                      class="form-group"
                    >
                      <div
                        class="react-select-container css-b62m3t-container"
                      >
                        <span
                          class="css-1f43avz-a11yText-A11yText"
                          id="react-select-2-live-region"
                        />
                        <span
                          aria-atomic="false"
                          aria-live="polite"
                          aria-relevant="additions text"
                          class="css-1f43avz-a11yText-A11yText"
                          role="log"
                        />
                        <div
                          class="react-select__control css-rn6d2z-control"
                        >
                          <div
                            class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
                          >
                            <div
                              class="react-select__placeholder css-1h3z2qk-placeholder"
                              id="react-select-2-placeholder"
                            >
                              Nothing selected
                            </div>
                            <div
                              class="react-select__input-container css-qbdosj-Input"
                              data-value=""
                            >
                              <input
                                aria-activedescendant=""
                                aria-autocomplete="list"
                                aria-describedby="react-select-2-placeholder"
                                aria-expanded="false"
                                aria-haspopup="true"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class="react-select__input"
                                id="react-select-2-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              class="react-select__indicator react-select__dropdown-indicator css-19wnsuc-indicatorContainer"
                            >
                              <svg
                                aria-hidden="true"
                                class="css-tj5bde-Svg"
                                focusable="false"
                                height="20"
                                viewBox="0 0 20 20"
                                width="20"
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="content-tab-form-left"
                >
                  <div
                    class="content-tab-title"
                  >
                    <h4>
                      Audio language*
                    </h4>
                  </div>
                  <div
                    class="headline-group"
                  >
                    <div
                      class="form-group"
                    >
                      <div
                        class="react-select-container css-b62m3t-container"
                      >
                        <span
                          class="css-1f43avz-a11yText-A11yText"
                          id="react-select-3-live-region"
                        />
                        <span
                          aria-atomic="false"
                          aria-live="polite"
                          aria-relevant="additions text"
                          class="css-1f43avz-a11yText-A11yText"
                          role="log"
                        />
                        <div
                          class="react-select__control css-rn6d2z-control"
                        >
                          <div
                            class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
                          >
                            <div
                              class="react-select__placeholder css-1h3z2qk-placeholder"
                              id="react-select-3-placeholder"
                            >
                              Select Audio
                            </div>
                            <div
                              class="react-select__input-container css-qbdosj-Input"
                              data-value=""
                            >
                              <input
                                aria-activedescendant=""
                                aria-autocomplete="list"
                                aria-describedby="react-select-3-placeholder"
                                aria-expanded="false"
                                aria-haspopup="true"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class="react-select__input"
                                id="react-select-3-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              class="react-select__indicator react-select__dropdown-indicator css-19wnsuc-indicatorContainer"
                            >
                              <svg
                                aria-hidden="true"
                                class="css-tj5bde-Svg"
                                focusable="false"
                                height="20"
                                viewBox="0 0 20 20"
                                width="20"
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="content-tab-form-split"
              >
                <div
                  class="content-tab-form-left"
                >
                  <div
                    class="content-tab-title"
                  >
                    <h4>
                      Close caption*
                    </h4>
                  </div>
                  <div
                    class="headline-group"
                  >
                    <div
                      class="form-group"
                    >
                      <div
                        class="react-select-container css-b62m3t-container"
                      >
                        <span
                          class="css-1f43avz-a11yText-A11yText"
                          id="react-select-4-live-region"
                        />
                        <span
                          aria-atomic="false"
                          aria-live="polite"
                          aria-relevant="additions text"
                          class="css-1f43avz-a11yText-A11yText"
                          role="log"
                        />
                        <div
                          class="react-select__control css-rn6d2z-control"
                        >
                          <div
                            class="react-select__value-container react-select__value-container--is-multi css-1fdsijx-ValueContainer"
                          >
                            <div
                              class="react-select__placeholder css-1h3z2qk-placeholder"
                              id="react-select-4-placeholder"
                            >
                              Select Close Caption
                            </div>
                            <div
                              class="react-select__input-container css-qbdosj-Input"
                              data-value=""
                            >
                              <input
                                aria-activedescendant=""
                                aria-autocomplete="list"
                                aria-describedby="react-select-4-placeholder"
                                aria-expanded="false"
                                aria-haspopup="true"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class="react-select__input"
                                id="react-select-4-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              class="react-select__indicator react-select__dropdown-indicator css-19wnsuc-indicatorContainer"
                            >
                              <svg
                                aria-hidden="true"
                                class="css-tj5bde-Svg"
                                focusable="false"
                                height="20"
                                viewBox="0 0 20 20"
                                width="20"
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="content-tab-form-left"
                >
                  <div
                    class="content-tab-title"
                  >
                    <h4>
                      Course category*
                    </h4>
                  </div>
                  <div
                    class="headline-group"
                  >
                    <div
                      class="form-group"
                    >
                      <div
                        class="react-select-container css-b62m3t-container"
                      >
                        <span
                          class="css-1f43avz-a11yText-A11yText"
                          id="react-select-5-live-region"
                        />
                        <span
                          aria-atomic="false"
                          aria-live="polite"
                          aria-relevant="additions text"
                          class="css-1f43avz-a11yText-A11yText"
                          role="log"
                        />
                        <div
                          class="react-select__control css-rn6d2z-control"
                        >
                          <div
                            class="react-select__value-container css-1fdsijx-ValueContainer"
                          >
                            <div
                              class="react-select__placeholder css-1h3z2qk-placeholder"
                              id="react-select-5-placeholder"
                            >
                              Select Category
                            </div>
                            <div
                              class="react-select__input-container css-qbdosj-Input"
                              data-value=""
                            >
                              <input
                                aria-activedescendant=""
                                aria-autocomplete="list"
                                aria-describedby="react-select-5-placeholder"
                                aria-expanded="false"
                                aria-haspopup="true"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class="react-select__input"
                                id="react-select-5-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              class="react-select__indicator react-select__dropdown-indicator css-19wnsuc-indicatorContainer"
                            >
                              <svg
                                aria-hidden="true"
                                class="css-tj5bde-Svg"
                                focusable="false"
                                height="20"
                                viewBox="0 0 20 20"
                                width="20"
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <div
        class="step-footer step-tab-pager"
      >
        <button
          class="btn btn-default steps_btn"
          data-direction="prev"
          style="display: none;"
        >
          PREVIOUS
        </button>
        <button
          class="btn btn-default steps_btn"
          data-direction="next"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
